prereise/gather/flexibilitydata/bus/bus_data.py

```diff
@@ -0,0 +1,339 @@
+        bus_pos[i, 0] = bus[i, 0]
+        bus_pos[i, 1] = sub[bus_sub_idx, 2]
+        bus_pos[i, 2] = sub[bus_sub_idx, 3]
+
+    with open(os.path.join(cache_path, "bus_pos.pkl"), "wb") as fh:
+        pkl.dump(bus_pos, fh)
+
+
+def get_bus_fips(bus_pos, cache_path, start_idx=0):
+    """Try to get FIPS of each bus in a case mat using FCC AREA API
+    Can take hours to run, save to cache file for future use
+
+    :param numpy.ndarray bus_pos: a list of (busID, lat, lon) stored as a numpy array
+    :param str cache_path: folder to store processed cache files
+    :param int start_idx: pointer to the index of a bus to start query from
+    """
+    bus_num = bus_pos.shape[0]
+    bus_fips_dict = {
+        "busid": bus_pos[:, 0],
+        "latitude": bus_pos[:, 1],
+        "longitude": bus_pos[:, 2],
+        "fips": [0] * bus_num,
+    }
+
+    url = "https://geo.fcc.gov/api/census/area"
+
+    for i in range(start_idx, bus_num):
+        if i % 1000 == 0:
+            with open(os.path.join(cache_path, "bus_fips.pkl"), "wb") as fh:
+                pkl.dump(bus_fips_dict, fh)
+
+        pos = bus_pos[i, :]
+
+        params = {"latitude": pos[1], "longitude": pos[2], "format": "json"}
+        r = requests.get(url, params=params)
+
+        assert r.status_code == 200
+
+        res = r.json()
+        if res["County"]["FIPS"] is not None:
+            bus_fips_dict["fips"][i] = int(res["County"]["FIPS"])
+        else:
+            bus_fips_dict["fips"][i] = -1
+
+        time.sleep(0.02)
+
+    with open(os.path.join(cache_path, "bus_fips.pkl"), "wb") as fh:
+        pkl.dump(bus_fips_dict, fh)
+
+
+def cleanup_zip(zipdict):
+    """Try to cleanup a zip dictionary obtained using online query by converting
+    to 5-digit integers. Several possible mis-format are considered
+
+    :param dict zipdict: a dictionary containing raw zip-code of buses
+    :return: (*dict*) -- a dictionary containing 5-digit zip codes
+    """
+    for i in range(len(zipdict["zip"])):
+        try:
+            zipdict["zip"][i] = int(zipdict["zip"][i])
+        except Exception:
+            if ":" in zipdict["zip"][i]:
+                zipdict["zip"][i] = int(zipdict["zip"][i].split(":")[0])
+
+            elif "-" in zipdict["zip"][i]:
+                zipdict["zip"][i] = int(zipdict["zip"][i].split("-")[0])
+
+            elif "‑" in zipdict["zip"][i]:
+                zipdict["zip"][i] = int(zipdict["zip"][i].split("‑")[0])
+
+            elif " " in zipdict["zip"][i]:
+                success = 0
+                for j in zipdict["zip"][i].split(" "):
+                    if len(j) == 5:
+                        zipdict["zip"][i] = int(j)
+                        success = 1
+                if success == 0:
+                    zipdict["zip"][i] = -1
+
+            else:
+                try:
+                    zipdict["zip"][i] = int(zipdict["zip"][i][0:5])
+                except Exception:
+                    zipdict["zip"][i] = -1
```

Comment:
it is not the case, many of them are like TX-77845, TX 77845 or TX:77845-1234

Reply:
In those cases, the current implementation `s.split("-")[0]` will extract "TX" out instead of the "77845", am I missing something?

Reply:
![image](https://user-images.githubusercontent.com/77214628/176496998-1d86f5b5-451e-44ad-8fb3-dfa7ca83b789.png)
I was giving an example of why we can't use the first 5 digit, I don't remember the details but from the screenshot it seems to consider TX 77845 type of thing.

Reply:
If we have "TX-77845" in the data, it will be picked up by the second `elif` before reaching the block you mentioned, right? Nonetheless, if we just want to match 5 consecutive digits in arbitrary strings and we are sure such pattern occur only once in a particular string, using regex might be the simplest way
```python
zipdict["zip"][i] = re.search("(?<!\d)\d{5}(?!\d)", zipdict["zip"][i]).group(0)
```